clients/input_copy.go

```diff
@@ -63,8 +70,62 @@ func (s *InputCopy) CopyInputToS3(requestID string, inputFile *url.URL) (inputVi
 		}
 	}
 
+	if encryptedKey != "" {
+		c, e := getFile(context.Background(), requestID, inputFile.String())
+
+		if e != nil {
+			glog.Errorf("error getting file: %w", err)
+			return
+		}
+
+		var decryptionKey *rsa.PrivateKey
+		if decryptionKey, err = crypto.LoadPrivateKey(VodDecryptPrivateKey); err != nil {
+			glog.Errorf("error loading private key: %w", err)
+			return
+		}
+
+		if decryptedFile, err = crypto.DecryptAESCBC(c, decryptionKey, encryptedKey); err != nil {
+			glog.Errorf("error decrypting file: %w", err)
+			return
+		}
+	}
+
+	if decryptedFile != nil {
+		var size int64
+		decryptedFileUrl := osTransferURL.String()
+
+		log.Log(requestID, "Copying decrypted file to S3", "source", inputFile.String(), "dest", decryptedFileUrl)
+		size, err = CopyReaderFile(context.Background(), decryptedFile, decryptedFileUrl, "", requestID)
+		if err != nil {
+			err = fmt.Errorf("failed to copy file(s): %w", err)
+			return
+		}
+		log.Log(requestID, "Copied", "bytes", size, "source", inputFile.String(), "dest", decryptedFileUrl)
+
+		signedURL, err = SignURL(osTransferURL)
+		if err != nil {
+			return
+		}
```

Comment:
Would be nice to reduce the code repetition here with the S3 copy below. It could be just 1 `if` on line 114 like
```go
if encryptedKey != "" {
  size, err = CopyFile(context.Background(), sourceOutputURL.String(), osTransferURL.String(), "", requestID, decryptFileReader(encryptedKey))
} else {
  size, err = CopyFile(context.Background(), sourceOutputURL.String(), osTransferURL.String(), "", requestID, nil) 
}
```